path_login_test.go

```diff
@@ -666,13 +670,59 @@ func testLogin_JWT(t *testing.T, jwks bool) {
 		}
 	}
 
-	// test auto notbefore from issue time
+	// test auto leeway from issue time
 	{
 		cl := jwt.Claims{
 			Subject:  "r3qXcK2bix9eFECzsU3Sbmh0K16fatW6@clients",
 			Issuer:   "https://team-vault.auth0.com/",
-			Expiry:   jwt.NewNumericDate(time.Now().Add(5 * time.Second)),
-			IssuedAt: jwt.NewNumericDate(time.Now().Add(-5 * time.Hour)),
+			IssuedAt: jwt.NewNumericDate(time.Now()),
+			Audience: jwt.Audience{"https://vault.plugin.auth.jwt.test"},
+		}
+
+		privateCl := struct {
+			User   string   `json:"https://vault/user"`
+			Groups []string `json:"https://vault/groups"`
+			Color  string   `json:"color"`
+		}{
+			"jeff",
+			[]string{"foo", "bar"},
+			"green",
+		}
+
+		jwtData, _ := getTestJWT(t, ecdsaPrivKey, cl, privateCl)
+
+		data := map[string]interface{}{
+			"role": "plugin-test",
+			"jwt":  jwtData,
+		}
+
+		req := &logical.Request{
+			Operation: logical.UpdateOperation,
+			Path:      "login",
+			Storage:   storage,
+			Data:      data,
+		}
+
+		resp, err := b.HandleRequest(context.Background(), req)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if resp == nil {
+			t.Fatal("got nil response")
+		}
+		if resp.IsError() {
+			t.Fatalf("unexpected error: %v", resp.Error())
+		}
+	}
+
+	// test custom expiration/notbefore from issue time
+	{
+		b, storage := setupBackend(t, false, false, true, false, false, jwks, true)
+
+		cl := jwt.Claims{
+			Subject:  "r3qXcK2bix9eFECzsU3Sbmh0K16fatW6@clients",
+			Issuer:   "https://team-vault.auth0.com/",
+			IssuedAt: jwt.NewNumericDate(time.Now()),
```

Comment:
To cover this change I think you'd want a couple more test cases. This one is a good/valid case (though a time the was near--but within--the limit would be better), but I don't see bad/invalid cases for when the current time is before the calculated nbf or after the calculated exp.